pkg/client/typed/discovery/discovery_client.go

```diff
@@ -212,6 +212,9 @@ func setDiscoveryDefaults(config *restclient.Config) error {
 	config.APIPath = ""
 	config.GroupVersion = nil
 	config.Codec = runtime.NoopEncoder{api.Codecs.UniversalDecoder()}
+	if config.UserAgent == "" {
```

Comment:
I need to remember to put that in coding-conventions so I can beat you over the head with it in the future.